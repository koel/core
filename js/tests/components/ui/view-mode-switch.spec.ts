import Component from '@/components/ui/view-mode-switch.vue'
import { shallow } from '@/tests/adapter'

describe('components/ui/view-mode-switch', () => {
  it('renders properly', () => {
    expect(shallow(Component, {
      propsData: {
        for: 'albums'
      }
    })).toMatchSnapshot()
  })

  it('changes the view mode', () => {
    const wrapper = shallow(Component, {
      propsData: {
        mode: 'list',
        for: 'albums'
      }
    })
    expect(wrapper.click('a.thumbnails').hasEmitted('viewModeChanged', 'thumbnails')).toBe(true)
    expect(wrapper.click('a.list').hasEmitted('viewModeChanged', 'list')).toBe(true)
  })
})
