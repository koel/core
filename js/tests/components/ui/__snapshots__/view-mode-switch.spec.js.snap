// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/view-mode-switch renders properly 1`] = `<span class="view-modes"><a title="View as thumbnails" role="button" href="#" class="thumbnails active"><i class="fa fa-th-large"></i></a> <a title="View as list" role="button" href="#" class="list"><i class="fa fa-list"></i></a></span>`;
